File: db-4.8.30/csharp/PreparedTransaction.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;
using BerkeleyDB.Internal;

namespace BerkeleyDB {
    /// <summary>
    /// A class representing a transaction that must be resolved by the
    /// application following <see cref="DatabaseEnvironment.Recover"/>.
    /// </summary>
    public class PreparedTransaction {
        private Transaction trans;
        private byte[] txnid;

        internal PreparedTransaction(DB_PREPLIST prep) {
            trans = new Transaction(prep.txn);
            txnid = prep.gid;
        }

        /// <summary>
        /// The transaction which must be committed, aborted or discarded.
        /// </summary>
        public Transaction Txn { get { return trans; } }
        /// <summary>
        /// The global transaction ID for the transaction. The global
        /// transaction ID is the one specified when the transaction was
        /// prepared. The application is responsible for ensuring uniqueness
        /// among global transaction IDs.
        /// </summary>
        public byte[] GlobalID { get { return txnid;}}
    }
}

File: db-4.8.30/java/src/com/sleepycat/db/PreparedTransaction.java
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 1999-2009 Oracle.  All rights reserved.
 *
 * $Id$
 */
package com.sleepycat.db;

import com.sleepycat.db.internal.DbTxn;

/**
The PreparedTransaction object is used to encapsulate a single prepared,
but not yet resolved, transaction.
*/
public class PreparedTransaction {
    private byte[] gid;
    private Transaction txn;

    PreparedTransaction(final DbTxn txn, final byte[] gid) {
        this.txn = new Transaction(txn);
        this.gid = gid;
    }

    public byte[] getGID() {
        return gid;
    }

    /**
    Return the transaction handle for the transaction.
    <p>
    @return
    The transaction handle for the transaction.
    */
    public Transaction getTransaction() {
        return txn;
    }
}
